pkg/roll/update_test.go

```diff
@@ -281,7 +297,7 @@ func SimulateRollingUpgradeDisable(t *testing.T, full, nonew bool) {
 		}
 
 		// calculate the next update
-		nextRemove, nextAdd := rollAlgorithm(len(old), len(new), target, target, minimum)
+		nextRemove, nextAdd := rollAlgorithm(len(old), len(new), target-len(new), len(new), target, minimum)
```

Comment:
I took the path of least resistance here, but I may need to reconsider either this test, or what `nextRemove` should mean in a mutable world